### PR TITLE
Use `assertEquals` instead of `assert`

### PR DESCRIPTION
**Pull request by danicheg**
This is pretty much a minor thing. However, we are using `MUnit` currently and can get nice diffs on assertions fails for free. I'm convinced `assert(foo == bar)` is almost anti-pattern and using `assertEquals(foo, bar)` is much better. If we agree on this, I'll add this point to the contributing guide. Perhaps, Scalafix-ninja may create a rewriting rule for this.
Also, we can replace `assert(boolExpr)` with `assert(boolExpr, true)`. But one may argue that this looks odd. I made it in several test classes to give you demonstrative examples. I think that nice diffs is worth the hassle anyway.

### CODE REVIEW

**Review by rossabaker — COMMENTED** [review]
Strong agree on `assertEquals(a, b)` over `assert(a == b)`.

I'm less convinced on the booleans.  They _must_ have returned false or thrown an exception we'll see anyway.  I guess there's a nice consistency to it, but I might be annoyed to fail a lint on it.  Here's the difference:

```
scala> assertEquals(false, true)
munit.ComparisonFailException: values are not the same
=> Obtained
false
=> Diff (- obtained, + expected)
-false
+true
  at munit.Assertions.failComparison(Assertions.scala:297)

scala> assert(false)
munit.FailException: assertion failed
  at munit.Assertions.fail(Assertions.scala:283)
```

**Comment by danicheg** [comment]
@rossabaker pretty reasonable point about using `assertEquals` for checking boolean expressions, so I've reverted these changes. This PR is ready for review.

**Comment by bplommer** [comment]
> Perhaps, Scalafix-ninja may create a rewriting rule for this.

Sure, add it to the list on the ticket 😄

File: tests/src/test/scala/org/http4s/DecodeSpec.scala [hunk 1]
```diff
@@ -95,50 +95,38 @@ class DecodeSpec extends Http4sSuite {
       // Not a valid first byte in UTF-8
       val source = Stream(0x80.toByte)
       val decoded = source.through(decode[Fallible](Charset.`UTF-8`)).compile.string
-      assert(decoded match {
-        case Left(_: MalformedInputException) => true
-        case _ => false
-      })
+      val Left(_: MalformedInputException) = decoded
```
Comment (bplommer) [inline tests/src/test/scala/org/http4s/DecodeSpec.scala:98]: This no longer has any assertion - what would a failure look like?

Comment (danicheg) [reply tests/src/test/scala/org/http4s/DecodeSpec.scala:98]: If matching fails the failure is `scala.MatchError: Right(_)(of class scala.util.Right)`.

**Review by rossabaker — APPROVED** [review]

**Comment by danicheg** [comment]
@bplommer do you have some points about this? Or can we move on?

**Comment by bplommer** [comment]
> @bplommer do you have some points about this? Or can we move on?

Go for it

**Review by bplommer — APPROVED** [review]

**Pull request merged**